hist/hist/src/TH1.cxx

```diff
@@ -2675,7 +2676,23 @@ TObject* TH1::Clone(const char* newname) const
       // when dictionary information is initialized, so we need to
       // keep obj->fFunction valid during its execution and
       // protect the update with the write lock.
+
+      // Avoid duplication of this histogram
+      auto oldstats = dynamic_cast<TVirtualPaveStats*>(fFunctions->FindObject("stats"));
+      TObject *oldparent = nullptr;
+      if (oldstats) {
+         oldparent = oldstats->GetParent();
+         oldstats->SetParent(nullptr);
+      }
+
       auto newlist = (TList*)fFunctions->Clone();
+
+      if (oldstats)
+         oldstats->SetParent(oldparent); // We could also use a const_cast of 'this'
```

Comment:
That comment is surprising reading the code: there is no mention or check that `oldstats->GetParent()` is `this`. Is this worth an assert?

Reply:
Probably.  The return value of GetParent "should be" ```this``` but 'technically this function (as is) doesn't care whether it is or it is not.  (However if one switch to setting it to this, you would need the assert).